src/mem/alloc.h

```diff
@@ -135,10 +133,19 @@ namespace snmalloc
     /**
      * Get the pagemap entry corresponding to a specific address.
      */
-    uint8_t get(void* p)
+    uint8_t get(uintptr_t p)
```

Comment:
@nwf, what do you think here?  My gut feeling is that we want this to be a non-provenance-carrying virtual address on CHERI, because we're going to derive a capability via the pagemap, so this probably wants to be `size_t` not `uintptr_t`?

Reply:
I agree that it probably should be something other than `uintptr_t`, but I am kind of wary of `size_t`, because that isn't required to be large enough to be an entire virtual address, merely the size of any particular object in the system.  I know segmented architectures are largely a thing of the past, but it would be nice to have a type specifically for "virtual address" even if it had to be defined in one of snmalloc's platform-specific utility headers.

If the choice is between `uintptr_t` and `size_t` specifically, I think I'd prefer the former: it's guaranteed to be big enough without being too big to comfortably pass (unlike, say, `uintmax_t`) and we can always shed the CHERI tags inside `get()` itself.

Reply:
So it sounds like we want an address_t, that is beyond what I wanted to do for now.